code/datums/mutations/adaptation.dm

```diff
@@ -0,0 +1,59 @@
+/datum/mutation/human/temperature_adaptation
+	name = "Temperature Adaptation"
+	desc = "A strange mutation that renders the host immune to damage from extreme temperatures. Does not protect from vacuums."
+	quality = POSITIVE
+	difficulty = 16
+	text_gain_indication = "<span class='notice'>Your body feels warm!</span>"
+	time_coeff = 5
+	instability = 25
+	conflicts = list(PRESSUREADAPT)
+
+/datum/mutation/human/temperature_adaptation/New(class_ = MUT_OTHER, timer, datum/mutation/human/copymut)
+	..()
+	if(!(type in visual_indicators))
+		visual_indicators[type] = list(mutable_appearance('icons/effects/genetics.dmi', "fire", -MUTATIONS_LAYER))
+
+/datum/mutation/human/temperature_adaptation/get_visual_indicator()
+	return visual_indicators[type][1]
+
+/datum/mutation/human/temperature_adaptation/on_acquiring(mob/living/carbon/human/owner)
+	if(..())
+		return
+	ADD_TRAIT(owner, TRAIT_RESISTCOLD, GENETIC_MUTATION)
+	ADD_TRAIT(owner, TRAIT_RESISTHEAT, GENETIC_MUTATION)
```

Comment:
I don't think this needs to make people immune to fire or intense heats, otherwise you're going to have people who can walk through plasma fires